File: frontend/apps/admin/src/services/file.service.ts
import type { FileService } from '#/generated/api/admin/service/v1/i_file.pb';
import type {
  CreateFileRequest,
  DeleteFileRequest,
  File,
  GetFileRequest,
  ListFileResponse,
  UpdateFileRequest,
} from '#/generated/api/file/service/v1/file.pb';
import type { Empty } from '#/generated/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/generated/api/pagination/v1/pagination.pb';

import { requestClient } from '#/utils/request';

/** 文件管理服务 */
class FileServiceImpl implements FileService {
  async Create(request: CreateFileRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/files', request);
  }

  async Delete(request: DeleteFileRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/files/${request.id}`);
  }

  async Get(request: GetFileRequest): Promise<File> {
    return await requestClient.get<File>(`/files/${request.id}`);
  }

  async List(request: PagingRequest): Promise<ListFileResponse> {
    return await requestClient.get<ListFileResponse>('/files', {
      params: request,
    });
  }

  async Update(request: UpdateFileRequest): Promise<Empty> {
    const id = request.data?.id;

    console.log('UpdateFile', request.data);
    if (request.data !== null && request.data !== undefined) {
      request.data.id = undefined;
    }
    return await requestClient.put<Empty>(`/files/${id}`, request);
  }
}

export const defFileService = new FileServiceImpl();

File: backend/app/admin/service/internal/service/file_service.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	pagination "github.com/tx7do/go-crud/api/gen/go/pagination/v1"
	"github.com/tx7do/go-utils/trans"
	"github.com/tx7do/kratos-bootstrap/bootstrap"
	"google.golang.org/protobuf/types/known/emptypb"

	"go-wind-admin/app/admin/service/internal/data"

	adminV1 "go-wind-admin/api/gen/go/admin/service/v1"
	fileV1 "go-wind-admin/api/gen/go/file/service/v1"

	"go-wind-admin/pkg/middleware/auth"
)

type FileService struct {
	adminV1.FileServiceHTTPServer

	log *log.Helper

	fileRepo *data.FileRepo
}

func NewFileService(ctx *bootstrap.Context, repo *data.FileRepo) *FileService {
	l := log.NewHelper(log.With(ctx.Logger, "module", "file/service/admin-service"))
	return &FileService{
		log:      l,
		fileRepo: repo,
	}
}

func (s *FileService) List(ctx context.Context, req *pagination.PagingRequest) (*fileV1.ListFileResponse, error) {
	return s.fileRepo.List(ctx, req)
}

func (s *FileService) Get(ctx context.Context, req *fileV1.GetFileRequest) (*fileV1.File, error) {
	return s.fileRepo.Get(ctx, req)
}

func (s *FileService) Create(ctx context.Context, req *fileV1.CreateFileRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("invalid parameter")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return nil, err
	}

	req.Data.CreatedBy = trans.Ptr(operator.UserId)

	if err = s.fileRepo.Create(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *FileService) Update(ctx context.Context, req *fileV1.UpdateFileRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("invalid parameter")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return nil, err
	}

	req.Data.UpdatedBy = trans.Ptr(operator.UserId)

	if err = s.fileRepo.Update(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *FileService) Delete(ctx context.Context, req *fileV1.DeleteFileRequest) (*emptypb.Empty, error) {
	if err := s.fileRepo.Delete(ctx, req); err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}
